libs/@local/harpc/client/typescript/src/net/internal/dns.ts

```diff
@@ -177,8 +149,57 @@ export const resolve = (
     }
 
     return Array.flatten(satisfying);
-  }).pipe(
-    Effect.tap((records) =>
-      logReverse(records).pipe(Effect.annotateLogs({ hostname, query })),
-    ),
-  );
+  });
+
+/** @internal */
+export const lookup = (
+  hostname: string,
+  query: {
+    records: NonEmptyReadonlyArray<RecordType>;
+  },
+) =>
+  Effect.gen(function* () {
+    const records = yield* Effect.tryPromise({
+      try: () => dns.lookup(hostname, { all: true }),
+      catch: (cause) => new DnsError({ cause }),
+    });
+
+    yield* Effect.fork(logEnvironment(hostname));
+
+    // partition into A and AAAA records
+    const [excluded, satisfying] = Array.partition(
+      records,
+      (record) => record.family === 4,
+    );
+
+    // we cannot determine the TTL of lookup records, therefore we set it to infinity
```

Comment:
Is this still planned in H-3793? If so, can we add a `TODO` or otherwise cancel the task?

Reply:
this is still planned, but the TTL on a lookup via `getaddrinfo` doesn't reveal a TTL. For that we'd need [H-3785](https://linear.app/hash/issue/H-3785/create-typescripteffect-dns-package). I can add a comment explaining this.

Reply:
added in 527c77f38